### ISSUE TITLE
Scenario Engine aplica cambios sobre features sin respetar el scaling del modelo GBR

### ISSUE THREAD

**Issue by barckcode** [barckcode/canary-tourism-demo#192]
## Descripción
El Scenario Engine modifica las features del modelo GBR (Gradient Boosting Regressor) directamente sin considerar que fueron normalizadas con `StandardScaler` durante el entrenamiento:

```python
# scenario_engine.py, líneas 233-240
feat_sc = feat.copy()
if occupancy_change_pct:
    feat_sc["room_occ_lag1"] *= (1 + occupancy_change_pct / 100)
    feat_sc["bed_occ_lag1"] *= (1 + occupancy_change_pct / 100)
```

Las features de entrenamiento pasaron por `StandardScaler.fit_transform()`, pero los cambios de escenario se aplican sobre los valores escalados como si fueran valores reales. Un cambio del +10% sobre un valor escalado no equivale a un +10% sobre el valor original.

Adicionalmente, al extender features para periodos futuros, los lags que no existen se rellenan con `fillna(0)` (línea ~243), lo que introduce un sesgo artificial: un lag de 0 arrivals no es "sin datos", es "cero turistas".

## Problema / Motivación
Este bug afecta directamente la utilidad del Scenario Engine como herramienta de planificación:

**Ejemplo concreto**: Un revenue manager del Gran Hotel Taoro (inaugurado 13/03/2026, 199 habitaciones) quiere simular: "¿Qué pasa si la ocupación sube un 10% por el efecto destino refugio del conflicto en Oriente Medio?"

Con el bug actual:
- El modelo recibe un valor escalado modificado que no corresponde a un +10% real
- La predicción resultante puede ser arbitrariamente incorrecta
- El usuario no sabe que la simulación es inválida

Con la corrección:
- El cambio del 10% se aplica sobre el valor real antes de re-escalar
- La predicción refleja correctamente el impacto del escenario

Con el Global MICE Forum 2026 en Tenerife (30 junio - 4 julio, ~250 profesionales), los escenarios deben ser fiables para planificar capacidad.

## Propuesta
1. **Inverse transform antes de modificar**: Deshacer el scaling, aplicar el cambio porcentual, re-escalar:
```python
real_values = scaler.inverse_transform(feat_sc[feature_cols])
real_values["room_occ_lag1"] *= (1 + occupancy_change_pct / 100)
feat_sc[feature_cols] = scaler.transform(real_values)
```

2. **Forward-fill en lugar de zero-fill** para lags faltantes: Usar el último valor conocido en lugar de 0
3. **Documentar las transformaciones** para que el revenue manager entienda qué está simulando

## Criterios de aceptación
- [ ] Los cambios de escenario se aplican sobre valores reales (pre-scaling), no sobre valores escalados
- [ ] Los lags faltantes se rellenan con forward-fill, no con cero
- [ ] Un escenario de +0% produce exactamente el mismo resultado que el baseline
- [ ] Un escenario de +10% ocupación produce un incremento razonable y proporcional en las predicciones
- [ ] Test: verificar que `predict_scenario(occupancy_change=0)` == `predict_scenario()` (sin cambios)

## Contexto adicional
- **Archivos afectados**: `backend/app/models/scenario_engine.py` (líneas 205-207, 233-243)
- **Modelo**: GBR (Gradient Boosting Regressor) con StandardScaler. Features: lags (1,3,6,12), rolling stats, ratio extranjeros, métricas de alojamiento, encoding estacional.
- **Referencia**: RoomRaccoon (competidor) lanzó RaccoonRev Plus con recomendaciones de tarifa basadas en escenarios correctamente calibrados. Duetto GameChanger aplica transformaciones inversas antes de simular.
- **Impacto MICE**: Tenerife celebró 1.367 congresos en 2024 con impacto de 394M€. Los escenarios deben poder simular el impacto de eventos MICE en la demanda.
